### PR TITLE
Added figure 7 to show imputation effectiveness

### PR DESCRIPTION
**Pull request by JacksonLChin**
- Adds figure 7, boxplots depicting differences between imputed
 and measured cytokine quantities

### CODE REVIEW

**Comment by JacksonLChin** [comment]
I know we discussed bar plots, but I thought these box-plots might give a better sense of the difference in distributions between cytokines and imputation methods

**Comment by aarmey** [comment]
Great start. You want to look at the correlation between measured and imputed here. This is because it doesn't just matter that the measured and imputed distributions are the same—you want to check that the values are the same even when looking at each subject.

**Comment by JacksonLChin** [comment]
Sure--would replicating the heatmaps in S4 make sense? Making heatmaps that show correlation between imputed and measured for both serum and plasma cytokines?

**Comment by aarmey** [comment]
You don't care about inter-cytokine correlations though. Only e.g., IL-2 actual to IL-2 inferred. That's why I thought a bar graph would make sense.

**Comment by JacksonLChin** [comment]
Here's a new commit that adds correlations between imputed and measured cytokines--generally seeing better correlations between imputed and measured cytokines than between measured cytokines from different sources, but there are a few exceptions.

**Comment by aarmey** [comment]
Are you assessing this only for those subjects with both serum and plasma measurements? Because otherwise I think these correlations are measuring different subsets of the data...

File: tfac/figures/figure7.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+    Parameters:
+        None
+
+    Returns:
+        tensor (numpy.array): cytokine measurements
+        imputed (numpy.array): cytokine measurements predicted via CMTF
+    """
+    tensor, matrix, patient_data = form_tensor()
+    both_cyto = patient_data.loc[:, 'type'].str.contains('0Serum1Plasma')
```
Comment (JacksonLChin) [inline tfac/figures/figure7.py:32]: Should be! This line should restrict only to patients with both serum and plasma data.

**Comment by aarmey** [comment]
I fixed one issue (we want to impute using the full dataset).

The improvement here is equivocal, so I'll need to think about this more. Let's leave this open.

**Comment by JacksonLChin** [comment]
@aarmey Here's an update that uses R^2 instead of correlation. We're seeing generally better results, but there's still some cytokines showing better accuracy when replaced with the alternate source rather than imputed via CMTF

**Comment by JacksonLChin** [comment]
I've also censored these to only include positive values--some of these cytokines have negative R2 scores that have been clipped.

**Pull request closed without merging**